iexec-sms-library/src/main/java/com/iexec/sms/api/SmsService.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2022 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.sms.api;
+
+import com.iexec.common.chain.WorkerpoolAuthorization;
+import com.iexec.common.tee.TeeWorkflowSharedConfiguration;
+import com.iexec.common.utils.BytesUtils;
+import feign.FeignException;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Optional;
+
+@Slf4j
+public class SmsService {
+
+    private final SmsClient smsClient;
+
+    private TeeWorkflowSharedConfiguration teeWorkflowConfiguration;
+
+    public SmsService(SmsClient smsClient) {
```

Comment:
If we start exposing an `SmsService`, we shall hide the `SmsClient`, since the connection to the SMS is handled by the service itself.

We might update the constructor to these
```
public SmsService(String smsUrl)

public SmsService(String smsUr, Logger.level levell)
```